core/systems/graphite/leaderboard_content_builder.ex

```diff
@@ -0,0 +1,224 @@
+          face: %{
+            type: :primary,
+            label: dgettext("eyra-assignment", "publish.button"),
+            bg_color: "bg-success"
+          }
+        },
+        icon: %{
+          action: publish_action,
+          face: %{type: :icon, icon: :publish, alt: dgettext("eyra-assignment", "preview.button")}
+        },
+        handle_click: &handle_publish/1
+      },
+      retract: %{
+        label: %{
+          action: retract_action,
+          face: %{
+            type: :secondary,
+            label: dgettext("eyra-assignment", "retract.button"),
+            text_color: "text-error",
+            border_color: "border-error"
+          }
+        },
+        icon: %{
+          action: retract_action,
+          face: %{
+            type: :icon,
+            icon: :retract,
+            alt: dgettext("eyra-benchmark", "assignment.button")
+          }
+        },
+        handle_click: &handle_retract/1
+      },
+      close: %{
+        label: %{
+          action: close_action,
+          face: %{
+            type: :primary,
+            label: dgettext("eyra-assignment", "close.button")
+          }
+        },
+        icon: %{
+          action: close_action,
+          face: %{type: :icon, icon: :close, alt: dgettext("eyra-assignment", "close.button")}
+        },
+        handle_click: &handle_close/1
+      },
+      open: %{
+        label: %{
+          action: open_action,
+          face: %{
+            type: :primary,
+            label: dgettext("eyra-assignment", "open.button")
+          }
+        },
+        icon: %{
+          action: open_action,
+          face: %{type: :icon, icon: :open, alt: dgettext("eyra-assignment", "open.button")}
+        },
+        handle_click: &handle_open/1
+      }
+    }
+  end
+
+  defp create_tabs(leaderboard, show_errors, assigns) do
+    get_tab_keys()
+    |> Enum.map(&create_tab(&1, leaderboard, show_errors, assigns))
+  end
+
+  defp get_tab_keys() do
+    [:settings, :download, :upload]
+  end
+
+  defp create_tab(:settings, leaderboard, show_errors, assigns) do
+    %{fabric: fabric, uri_origin: uri_origin, viewport: viewport, breakpoint: breakpoint} =
+      assigns
+
+    child =
+      Fabric.prepare_child(fabric, :settings_form, Graphite.LeaderboardSettingsView, %{
+        entity: leaderboard,
+        uri_origin: uri_origin,
+        viewport: viewport,
+        breakpoint: breakpoint
+      })
+
+    %{
+      id: "settings_form",
+      ready: false,
+      show_errors: show_errors,
+      title: "Settings",
+      forward_title: "Change settings",
+      backward_title: "Change settings",
+      type: :fullpage,
+      child: child
+    }
+  end
+
+  defp create_tab(:download, leaderboard, show_errors, assigns) do
+    %{fabric: fabric, uri_origin: uri_origin, viewport: viewport, breakpoint: breakpoint} =
+      assigns
+
+    child =
+      Fabric.prepare_child(fabric, :download_form, Graphite.LeaderboardDownloadView, %{
```

Comment:
In the future the upload / download tab might be renamed to submissions / scores since no manual upload/download will take place. Not sure though if naming them already submissions / scores is the best way to go. So this is just meant as a consideration for you.